pkg/settlement/swap/chequebook/factory.go

```diff
@@ -121,12 +113,25 @@ func (c *factory) VerifyBytecode(ctx context.Context) (err error) {
 
 // VerifyChequebook checks that the supplied chequebook has been deployed by this factory.
 func (c *factory) VerifyChequebook(ctx context.Context, chequebook common.Address) error {
-	deployed, err := c.instance.DeployedContracts(&bind.CallOpts{
-		Context: ctx,
-	}, chequebook)
+	callData, err := factoryABI.Pack("deployedContracts", chequebook)
+	if err != nil {
+		return err
+	}
+
+	output, err := c.transactionService.Call(ctx, &transaction.TxRequest{
+		To:   &c.address,
+		Data: callData,
+	})
+	if err != nil {
+		return err
+	}
+
+	results, err := factoryABI.Unpack("deployedContracts", output)
 	if err != nil {
 		return err
 	}
+
+	deployed := *abi.ConvertType(results[0], new(bool)).(*bool)
```

Comment:
this can potentially panic

Reply:
yes, but that is how those abi functions are supposed to be used, it is exactly the same code as you would find in the bindings. the fact the unpack did not error should guarantee that this conversion will work. Also `abi.ConvertType` already panics internally if the cast is not possible, so even if we checked something here it would already be too late.

Reply:
I added some basic sanity checks for now.